aptos-move/framework/aptos-framework/sources/configs/staking_config.move

```diff
@@ -0,0 +1,193 @@
+/// Provides the configuration for staking and rewards
+module aptos_framework::staking_config {
+    use std::error;
+    use aptos_framework::system_addresses;
+
+    /// Invalid required stake lockup value.
+    const EINVALID_LOCKUP_VALUE: u64 = 1;
+    /// Invalid rewards rate.
+    const EINVALID_REWARDS_RATE: u64 = 2;
+    /// Invalid required stake range, usually happens if min > max.
+    const EINVALID_STAKE_RANGE: u64 = 3;
+
+    /// Validator set configurations that will be stored with the @aptos_framework account.
+    struct StakingConfig has copy, drop, key {
+        // A validator needs to stake at least this amount to be able to join the validator set.
+        // If after joining the validator set and at the start of any epoch, a validator's stake drops below this amount
+        // they will be removed from the set.
+        minimum_stake: u64,
+        // A validator can only stake at most this amount. Any larger stake will be rejected.
+        // If after joining the validator set and at the start of any epoch, a validator's stake exceeds this amount,
+        // their voting power and rewards would only be issued for the max stake amount.
+        maximum_stake: u64,
+        recurring_lockup_duration_secs: u64,
+        // Whether validators are allow to join/leave post genesis.
+        allow_validator_set_change: bool,
+        // The maximum rewards given out every epoch. This will be divided by the rewards rate denominator.
+        // For example, 0.001% (0.00001) can be represented as 10 / 1000000.
+        rewards_rate: u64,
+        rewards_rate_denominator: u64,
+    }
+
+    public fun initialize(
+        aptos_framework: &signer,
+        minimum_stake: u64,
+        maximum_stake: u64,
+        recurring_lockup_duration_secs: u64,
+        allow_validator_set_change: bool,
+        rewards_rate: u64,
+        rewards_rate_denominator: u64,
+    ) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+
+        // This can fail genesis but is necessary so that any misconfigurations can be corrected before genesis succeeds
+        validate_required_stake(minimum_stake, maximum_stake);
+
+        assert!(
+            rewards_rate_denominator > 0,
+            error::invalid_argument(EINVALID_REWARDS_RATE),
+        );
+
+        move_to(aptos_framework, StakingConfig {
+            minimum_stake,
+            maximum_stake,
+            recurring_lockup_duration_secs,
+            allow_validator_set_change,
+            rewards_rate,
+            rewards_rate_denominator,
+        });
+    }
+
+    public fun get(): StakingConfig acquires StakingConfig {
+        *borrow_global<StakingConfig>(@aptos_framework)
+    }
+
+    /// Return whether validator set changes are allowed
+    public fun get_allow_validator_set_change(config: &StakingConfig): bool {
```

Comment:
This would be so much better once we have the ability to make a resource "public" so we wouldn't need all of these getters to allow someone to read the content of a struct/resource they already have on hand.